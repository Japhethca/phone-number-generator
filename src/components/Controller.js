import React from 'react';
import {css} from 'glamor';

const controllerStyles = css({
  display: 'flex',
  alignItems: 'center',
  justifyContent: 'center',
});

const buttonStyles = css({
  padding: '10px',
  position: 'relative',
  fontSize: '12pt',
  marginLeft: '10px',
  borderRadius: '5px',
  fontWeight: 'bold',
  backgroundColor: '#00A550',
  border: 0,
  ':hover': {
    backgroundColor: '	#006400'
  }
});

const inputStyles = css({
  padding: '10px',
  outline: 0,
  border: 0,
  minWidth: '150px',
  borderRadius: '10px'
});

const sortOrderSelectStyles = css({
  marginLeft: '20px',
  padding: '5px',
  fontSize: '12pt',
});

export default function Controller({ 
  getNumbers, 
  generateNumbers, 
  noOfNumbers, 
  setNoOfNumbers,
}) {
  const getOrGenerateNumbers = (noOfNumbers) => {
    return noOfNumbers ? generateNumbers(noOfNumbers) : getNumbers();
  };

  const selectChange = (event) => {
    const selectedSortValue = event.target.value;
    return getNumbers(selectedSortValue);
  };

  const handleInputChange = (event) => setNoOfNumbers(event.target.value);
  return (
    <div {...controllerStyles}>
      <form onSubmit={(e) => {
          e.preventDefault();
          getOrGenerateNumbers(noOfNumbers);
        }}>
        <input {...inputStyles} 
          type="number"  
          value={noOfNumbers} 
          onChange={handleInputChange}
          placeholder="size of numbers"
          max="10000"
          min="0"
        />
        <button id="button" {...buttonStyles}>
          Generate Numbers
        </button>
      </form>
      <div>
        <select onChange={selectChange} name="sortOrder" id="" {...sortOrderSelectStyles}>
          <option value="">Sort Order</option>
          <option value="ASC">Ascending</option>
          <option value="DESC">Descending</option>
        </select>
      </div>
    </div>
  );
}