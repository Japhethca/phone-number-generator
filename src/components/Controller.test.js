import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';

import Controller from './Controller';

describe('<Controller />', () => {
  const props = {
    setNoOfNumbers: jest.fn(),
    noOfNumbers: 10,
    getNumbers: jest.fn(),
    generateNumbers: jest.fn(),
  };

  const getComponent = (newProps = props) => {
    const component = <Controller {...newProps}/>;
    return {
      component: mount(component),
      tree: renderer.create(component).toJSON(),
    };
  };

  it('should render successfully', () => {

    const { tree } = getComponent();
    expect(tree).toMatchSnapshot();
  });

  it('should generate numbers when button is clicked', () => {
    const { component } = getComponent();
    const form = component.find('form');
    form.simulate('submit');
    expect(props.generateNumbers).toHaveBeenCalledWith(props.noOfNumbers);
  });

  it('should get generated numbers when button is clicked', () => {
    const newProps = {
      ...props,
      noOfNumbers: 0,
    };
    const { component } = getComponent(newProps);
    const form = component.find('form');
    form.simulate('submit');
    expect(props.getNumbers).toHaveBeenCalled();
  });

  it('should handle input change', () => {
    const { component } = getComponent();
    const input = component.find('input');
    input.simulate('change', { target: { value: 5 }});
    expect(props.setNoOfNumbers).toHaveBeenCalledWith(5);
  });

  it('should handle sort order selection', () => {
    const { component } = getComponent();
    const selectInput = component.find('select');
    selectInput.simulate('change', { target: { value: 'ASC' }});
    expect(props.getNumbers).toHaveBeenCalled();
  });
});
