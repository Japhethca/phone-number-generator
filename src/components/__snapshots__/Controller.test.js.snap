// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Controller /> should render successfully 1`] = `
<div
  data-css-1ugxpw4=""
>
  <input
    data-css-1oss144=""
    maxLength="8"
    onChange={[Function]}
    placeholder="size of numbers"
    required={true}
    type="number"
    value={10}
  />
  <button
    data-css-fs9jv2=""
    id="button"
    onClick={[Function]}
  >
    Generate Numbers
  </button>
  <div>
    <select
      data-css-1i5es5r=""
      id=""
      name="sortOrder"
      onChange={[Function]}
    >
      <option
        value=""
      >
        Sort Order
      </option>
      <option
        value="ASC"
      >
        Ascending
      </option>
      <option
        value="DESC"
      >
        Descending
      </option>
    </select>
  </div>
</div>
`;
